ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj

```diff
@@ -211,14 +211,34 @@
 (defmethod add-s3-url :default
   [context concept urls]
   (errors/throw-service-errors
-   :invalid-data [(format "Add s3 url is not supported for format [%s]" (:format concept))]))
+   :invalid-data [(format "Adding s3 urls is not supported for format [%s]" (:format concept))]))
+
+(defmulti append-s3-url
+  "Add s3 url to the given granule concept."
+  (fn [context concept urls]
+    (mt/format-key (:format concept))))
+
+(defmethod append-s3-url :echo10
+  [context concept urls]
+  (s3-echo10/append-s3-url concept urls))
+
+(defmethod append-s3-url :umm-json
+  [context concept urls]
+  (update-umm-g-urls concept urls s3-umm-g/append-s3-url))
+
+(defmethod append-s3-url :default
+  [context concept urls]
+  (errors/throw-service-errors
+   :invalid-data [(format "Appending s3 urls is not supported for format [%s]" (:format concept))]))
 
 (defmulti update-granule-concept
   "Perform the update of the granule concept."
   (fn [context concept bulk-update-params user-id]
     (keyword (:event-type bulk-update-params))))
 
-(defmethod update-granule-concept :update_field:opendaplink
+(defn- update-granule-opendap-fields
+  "For OPeNDAP links, there may be at most 2 urls, 1 cloud, 1 on-prem.
+  therefore update and append bulk granule operations are identical."
```

Comment:
I'm not sure I agree with this behavior. If the operations are identical, what's the point? It seems to add unneeded complexity. 

The use case for `append` in my eyes is "I want to add a link, but make sure I don't overwrite my existing links." For that reason, I think two alternatives are A) don't support append for OPeNDAP links, or B) have `append` fail if a link is added for an OPeNDAP link type which is already populated.

Reply:
Could definitely do that. Need to have a discussion about the content of the ticket to finalize but making that change won't be difficult.